src/java/us/kbase/mobu/tester/test/ModuleTesterFailHardTest.java

```diff
@@ -0,0 +1,113 @@
+import us.kbase.mobu.ModuleBuilder;
+import us.kbase.mobu.initializer.ModuleInitializer;
+import us.kbase.mobu.initializer.test.DockerClientServerTester;
+import us.kbase.mobu.tester.ModuleTester;
+import us.kbase.scripts.test.TestConfigHelper;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyInt;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+/** This class tests to ensure that the unit test run fails hard if the callback
+ *  server cannot be instantiated.
+ *
+ */
+@PowerMockIgnore({"javax.net.ssl.*"})   // PowerMockRunner tries to classload a different set of net classes
+@RunWith( PowerMockRunner.class )
+@PrepareForTest( CallbackServer.class )
+public class ModuleTesterFailHardTest {
+    private static final String SIMPLE_MODULE_NAME = "ASimpleModule_for_unit_testing";
+    private static final boolean cleanupAfterTests = true;
+    private static List<String> createdModuleNames = new ArrayList<String>();
+    private static AuthToken token;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        token = TestConfigHelper.getToken();
+        PowerMockito.mockStatic(CallbackServer.class);
+        // force function to always return null
+        when(CallbackServer.getCallbackUrl(anyInt(),any(String[].class))).thenReturn(null);
+        Assert.assertNull(CallbackServer.getCallbackUrl(0, new String[1]));
+    }
+
+    @AfterClass
+    public static void tearDownModule() throws Exception {
+        if (cleanupAfterTests)
+            for (String moduleName : createdModuleNames)
+                try {
+                    deleteDir(moduleName);
+                } catch (Exception ex) {
+                    System.err.println("Error cleaning up module [" +
+                            moduleName + "]: " + ex.getMessage());
+                }
+    }
+
+    @After
+    public void afterText() {
+        System.out.println();
+    }
+
+    private static void deleteDir(String moduleName) throws Exception {
+        File module = new File(moduleName);
+        if (module.exists() && module.isDirectory())
+            FileUtils.deleteDirectory(module);
+    }
+
+    private void init(String lang, String moduleName) throws Exception {
+        deleteDir(moduleName);
+        createdModuleNames.add(moduleName);
+        ModuleInitializer initer = new ModuleInitializer(moduleName, token.getUserName(),
+                lang, false);
+        initer.initialize(true);
+    }
+
+    private int runTestsInDocker(String moduleName) throws Exception {
+        File moduleDir = new File(moduleName);
+        return runTestsInDocker(moduleDir, token);
+    }
+
+    public static int runTestsInDocker(File moduleDir, AuthToken token) throws Exception {
+        return runTestsInDocker(moduleDir, token, false);
+    }
+
+    public static int runTestsInDocker(File moduleDir, AuthToken token,
```

Comment:
Is it really necessary to set up all the docker stuff? I assumed the failures would happen before the docker image was required, but I'm not sure about that.

Reply:
Regarding build failure, you may need to do a git fetch and merge of your jars dir since we recently added powermock to the jars. 

ModuleTesterTest.java in part tests the successful setup of the callback server. ModuleTesterFailHardTest.java is a complement test to ModuleTesterTest.java to show that the system fails immediately when the exact same test is executed with a failed setup of callback server.